sw/device/tests/aon_timer_smoketest.c

```diff
@@ -44,24 +35,14 @@ static void aon_timer_test_wakeup_timer(dif_aon_timer_t *aon) {
 }
 
 static void aon_timer_test_watchdog_timer(dif_aon_timer_t *aon) {
-  // Make sure that watchdog timer is stopped.
-  CHECK_DIF_OK(dif_aon_timer_watchdog_stop(aon));
-
-  // Make sure the watchdog IRQ is cleared to avoid false positive.
-  CHECK_DIF_OK(
-      dif_aon_timer_irq_acknowledge(aon, kDifAonTimerIrqWdogTimerBark));
-  bool is_pending;
-  CHECK_DIF_OK(dif_aon_timer_irq_is_pending(aon, kDifAonTimerIrqWdogTimerBark,
-                                            &is_pending));
-  CHECK(!is_pending);
-
   // Test the watchdog timer functionality by setting a single cycle "bark"
   // counter. Delay to compensate for AON Timer 200kHz clock (less should
   // suffice, but to be on a cautious side - lets keep it at 100 for now).
-  CHECK_DIF_OK(dif_aon_timer_watchdog_start(aon, 1, 0xffffffff, false, false));
+  aon_timer_testutils_watchdog_config(aon, 1, 0xffffffff);
   usleep(100);
 
   // Make sure that the timer has expired.
+  bool is_pending;
   CHECK_DIF_OK(dif_aon_timer_irq_is_pending(aon, kDifAonTimerIrqWdogTimerBark,
                                             &is_pending));
   CHECK(is_pending);
```

Comment:
Alternately, you could use the spinwait macro. Here and above.

Reply:
I don't like this... and perhaps I don't understand it: The watchdog will cause a reset, which resets the processor, so this code may never be run. The way other tests do is to check if they came from POR, or if the reset_info is different, in this case it would set a bit corresponding to an aon_timer reset. These checks should be performed in that "branch" of the code when it starts up.